packages/rabbitmq/src/TxSubmitWorker.ts

```diff
@@ -0,0 +1,273 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import { Channel, Connection, Message, connect } from 'amqplib';
+import { Logger, dummyLogger } from 'ts-log';
+import { ProviderError, ProviderFailure, TxSubmitProvider } from '@cardano-sdk/core';
+import { TX_SUBMISSION_QUEUE } from './rabbitmqTxSubmitProvider';
+
+/**
+ * Configuration options parameters for the TxSubmitWorker
+ */
+export interface TxSubmitWorkerOptions {
+  /**
+   * The logger. Default: dummyLogger (no log at all)
+   */
+  logger?: Logger;
+
+  /**
+   * Instructs the worker to process multiple transactions simultaneously.
+   * Default: false (serial mode)
+   */
+  parallel?: boolean;
+
+  /**
+   * The RabbitMQ polling cycle in milliseconds.
+   * Ignored in parallel mode. Default: 500
+   */
+  pollingCycle?: number;
+
+  /**
+   * The RabbitMQ connection URL
+   */
+  rabbitmqURL: URL;
+
+  /**
+   * The provider to use to submit tx
+   */
+  txSubmitProvider: TxSubmitProvider;
+}
+
+/**
+ * Controller class for the transactions submission worker which gets
+ * transactions from RabbitMQ and submit them via the TxSubmitProvider
+ */
+export class TxSubmitWorker {
+  /**
+   * The RabbitMQ channel
+   */
+  #channel?: Channel;
+
+  /**
+   * The RabbitMQ connection
+   */
+  #connection?: Connection;
+
+  /**
+   * Flag to exit from the infinite loop in case of sequential tx handlig
+   */
+  #continueForever = false;
+
+  /**
+   * The RabbitMQ consumerTag
+   */
+  #consumerTag?: string;
+
+  /**
+   * The function to call to resolve the start method exit Promise
+   */
+  #exitResolver?: () => void;
+
+  /**
+   * The logger
+   */
+  #logger: Logger;
+
+  /**
+   * Worker configuration: it was configured to process multiple tx in parallel
+   */
+  #parallel?: boolean;
+
+  /**
+   * The RabbitMQ polling cycle in milliseconds.
+   */
+  #pollingCycle: number;
+
+  /**
+   * Worker configuration: the RabbitMQ connection URL
+   */
+  #rabbitmqURL: URL;
+
+  /**
+   * The internal worker status
+   */
+  #status: 'connected' | 'connecting' | 'error' | 'idle' = 'idle';
+
+  /**
+   * The TxSubmitProvider instance
+   */
+  #txSubmitProvider: TxSubmitProvider;
+
+  /**
+   * @param {TxSubmitWorkerOptions} options The configuration options
+   */
+  constructor(options: TxSubmitWorkerOptions) {
+    const { logger, parallel, pollingCycle, rabbitmqURL, txSubmitProvider } = options;
+
+    this.#logger = logger || dummyLogger;
+    this.#parallel = parallel;
+    this.#pollingCycle = pollingCycle || 500;
+    this.#rabbitmqURL = rabbitmqURL;
+    this.#txSubmitProvider = txSubmitProvider;
+  }
+
+  /**
+   * Get the status of the worker
+   *
+   * @returns the status of the worker
+   */
+  getStatus() {
```

Comment:
The drawback of making `#status` public is that it can be changed by the `TxSubmitWorker` user.
That's just a way to give the `TxSubmitWorker` user a RO way to access the status of the worker.
Please just confirm it is preferred to be public, and I'll change it.

Reply:
```suggestion
  get status() {
```